benches/socket.rs

```diff
@@ -106,3 +108,44 @@ fn bench_transfer_one_megabyte(b: &mut Bencher) {
     });
     b.bytes = len as u64;
 }
+
+#[bench]
+fn bench_transfer_one_megabyte_random_chunks(b: &mut Bencher) {
+    let len = 1024 * 1024;
+    let server_addr = next_test_ip4();
+    let data = (0..len).map(|x| x as u8).collect::<Vec<u8>>();
+    let data_arc = Arc::new(data);
+    let mut received: Vec<u8> = Vec::with_capacity(len);
+
+    b.iter(|| {
+        let data = data_arc.clone();
+        let mut server = iotry!(UtpSocket::bind(server_addr));
+
+        thread::spawn(move || {
+            let mut written_total = 0;
+            let mut client = iotry!(UtpSocket::connect(server_addr));
+            while written_total < len {
+                let chunk_size = rand::random::<u16>() as usize + 1;
+                let chunk_size = min(chunk_size, len - written_total);
+                //println!("written_total == {}; chunk_size == {}", written_total, chunk_size);
+                let written = iotry!(client.send_to(&data[written_total..(written_total + chunk_size)]));
```

Comment:
I'm not sure it's wise to benchmark like this. Each run will have very different characteristics. Can you change code in a way that we can at least control the seed? This way we can run the benchmark against two different implementations to see which one perform better under the same seed.